### ISSUE TITLE
Debugging Not Working on .NET 4.8 Isolated Azure Functions Project both in VS 2022 and VS Code IDEs

### ISSUE THREAD

**Issue by HariKrishna598** [Azure/azure-functions-dotnet-worker#1171]
**Steps to Reproduce**:
1. Created the .NET 4.8 V4 Isolated Azure Functions Project - basic Http Trigger
2. without debugging working fine from VS IDE but not from VS 2022
![2022-11-10 18_51_37-](https://user-images.githubusercontent.com/88704316/201110108-94701f9b-020d-4750-a999-b33c46595f81.png)

4. Running the function with debugging is not working - debugger points are not hitting the code.

Following this [MS Doc](https://learn.microsoft.com/en-us/azure/azure-functions/dotnet-isolated-process-guide#debugging-when-targeting-net-framework),

- Kept debugging points at the code `FunctionsDebugger.Enable();` and also in HttpTrigger Function Code.
- Not Sure which process to be attached for debugging the .NET 4.8 Functions Isolated one.

- But tried this cmdlet `func host start --dotnet-isolated-debug` in the VS Code Terminal of the same project - without hitting the breakpints, it directly showing the output

Please Assist!

**Comment by kshyju** [comment]
@HariKrishna598 Can you confirm you have the latest version of Visual studio? [17.4.0 is the latest as of today](https://devblogs.microsoft.com/visualstudio/visual-studio-2022-17-4/).

Also what is the [core-tools](https://learn.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=v4%2Cwindows%2Ccsharp%2Cportal%2Cbash) version you have in your machine?

**Comment by HariKrishna598** [comment]
> @HariKrishna598 Can you confirm you have the latest version of Visual studio? [17.4.0 is the latest as of today](https://devblogs.microsoft.com/visualstudio/visual-studio-2022-17-4/).
> 
> Also what is the [core-tools](https://learn.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=v4%2Cwindows%2Ccsharp%2Cportal%2Cbash) version you have in your machine?

VS 2022 Version 17.4.0, 64-bit
AF Core Tools Version 4

**Comment by kshyju** [comment]
Just to confirm, do you have the latest worker packages? [In this issue](https://github.com/Azure/azure-functions-dotnet-worker/issues/1045#issuecomment-1252290057), that fixed a similar issue for you.

**Comment by HariKrishna598** [comment]
> Just to confirm, do you have the latest worker packages? [In this issue](https://github.com/Azure/azure-functions-dotnet-worker/issues/1045#issuecomment-1252290057), that fixed a similar issue for you.

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net48</TargetFramework>
    <AzureFunctionsVersion>v4</AzureFunctionsVersion>
    <OutputType>Exe</OutputType>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Azure.Functions.Worker" Version="1.8.0" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Http" Version="3.0.13" />
    <PackageReference Include="Microsoft.Azure.Functions.Worker.Sdk" Version="1.7.0" />
  </ItemGroup>
  <ItemGroup>
    <None Update="host.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </None>
    <None Update="local.settings.json">
      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
      <CopyToPublishDirectory>Never</CopyToPublishDirectory>
    </None>
  </ItemGroup>
  <ItemGroup>
    <Folder Include="Properties\" />
  </ItemGroup>
</Project>
```

This is my `.csproj` in .NET 48 v4 Azure Functions Project

**Comment by kshyju** [comment]
I suspect your visual studio instance failed to refresh the azure functions tools automatically. You can manually do this refresh by following the instructions [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/989#issuecomment-1283018013). Let us know if that solves your problem.

**Comment by HariKrishna598** [comment]
> I suspect your visual studio instance failed to refresh the azure functions tools automatically. You can manually do this refresh by following the instructions [here](https://github.com/Azure/azure-functions-dotnet-worker/issues/989#issuecomment-1283018013). Let us know if that solves your problem.

After manually updating the Azure Functions by following the instructions in above link, it worked (both debugging and running).

1. Do we need to update it manually every time when it fails (What is the root cause of failing the automatic update in VS IDE)?
2. What about the debugging fix in VS Code for the same issue?

**Comment by kshyju** [comment]
@HariKrishna598 

> _Do we need to update it manually every time when it fails (What is the root cause of failing the automatic update in VS IDE)?_

No, you do not need to do that. Typically Visual studio will automagically update it for you. Looks like there was an issue on the VS side which prevented it. We have notified the VS team about this and they are looking into it.

> _What about the debugging fix in VS Code for the same issue?_

Can you confirm what version of core tools you have in your machine? VS code uses the core tools installed in your machine. You can get this info by running `func` command. The result will have both core tools version and functions runtime version. If you are not on the latest core tools, I suggest [upgrading to the latest](https://learn.microsoft.com/en-us/azure/azure-functions/functions-run-local?tabs=v4%2Cwindows%2Ccsharp%2Cportal%2Cbash#v2) and try again. Also, ensure you are using the latest worker nuget packages(in your .csproj file). 

Let us know if you are still running into issues with this.

**Comment by HariKrishna598** [comment]
`func --version` > 4.0.4785 is the current version installed in my pc.
